lib/pure/asyncnet.nim

```diff
@@ -149,7 +149,7 @@ proc newAsyncSocket*(fd: AsyncFD, domain: Domain = AF_INET,
   ## async dispatcher. You need to do this manually. If you have used
   ## `newAsyncNativeSocket` to create `fd` then it's already registered.
   assert fd != osInvalidSocket.AsyncFD
-  new(result)
+  unsafeNew(result, sizeof(AsyncSocketDesc) + (if buffered: BufferSize else: 0))
```

Comment:
How much of a performance impact would using a `seq` have here over this unsafe code? if it's not significant you should consider it

Reply:
From the point of view of performance, there should be no significant difference, but it can reduce the memory management pressure when the number of sockets is large.

Reply:
Wouldn't memory management pressure affect performance?

I think whatever way it can be done without using "unsafe" operations is good.

Reply:
Agree, I'll change it